[Bug][config-ui] Blueprint runs don't show subtasks

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.


### What happened

Blueprint runs don't show subtasks, I have no idea about the progress. After an hour's running, the run failed, which was frustrating.

![image](https://user-images.githubusercontent.com/14050754/192822807-c472dac8-c6bb-4063-8a93-91664d1ae140.png)


### What you expected to happen

Something like this:
![image](https://user-images.githubusercontent.com/14050754/192822981-4b9ed749-a082-48be-83f9-ccd1fe1f56d9.png)


### How to reproduce

Visit [here](https://ui-my-svr02.demo.devlake.io/) or your own config-ui and create a blueprint

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

logs:
```
time="2022-07-26 10:35:28" level=info msg="endpoint DELETE /plugins/ae/connections/:connectionId github.com/apache/incubator-devlake/api.RegisterRouter.func1 5"
[GIN-debug] [WARNING] You trusted all proxies, this is NOT safe. We recommend you to set a value.
Please check https://pkg.go.dev/github.com/gin-gonic/gin#readme-don-t-trust-all-proxies for details.
[GIN-debug] Listening and serving HTTP on :8080
time="2022-07-26 10:35:29" level=info msg=" [db] /app/services/pipeline.go:215 Error 1053: Server shutdown in progress
[0.283ms] [rows:0] SELECT * FROM `_devlake_pipelines` WHERE status = 'TASK_CREATED' ORDER BY id ASC LIMIT 1"
[mysql] 2022/07/26 10:35:30 packets.go:123: closing bad idle connection: EOF
[mysql] 2022/07/26 10:35:30 connection.go:173: driver: bad connection
time="2022-07-26 10:35:30" level=info msg=" [db] /app/services/pipeline.go:215 dial tcp 10.43.18.27:3306: connect: connection refused
[0.804ms] [rows:0] SELECT * FROM `_devlake_pipelines` WHERE status = 'TASK_CREATED' ORDER BY id ASC LIMIT 1"
time="2022-07-26 10:35:31" level=info msg=" [db] /app/services/pipeline.go:215 dial tcp 10.43.18.27:3306: connect: connection refused
[0.585ms] [rows:0] SELECT * FROM `_devlake_pipelines` WHERE status = 'TASK_CREATED' ORDER BY id ASC LIMIT 1"
time="2022-07-26 10:35:32" level=info msg=" [db] /app/services/pipeline.go:215 dial tcp 10.43.18.27:3306: connect: connection refused
[0.616ms] [rows:0] SELECT * FROM `_devlake_pipelines` WHERE status = 'TASK_CREATED' ORDER BY id ASC LIMIT 1"

```

--- a/services/task.go
+++ b/services/task.go
@@ -280,16 +280,15 @@ func runTaskStandalone(parentLog core.Logger, taskId uint64) errors.Error {
 }
 
 func updateTaskProgress(taskId uint64, progress chan core.RunningProgress) {
-	runningTasks.mu.Lock()
-	defer runningTasks.mu.Unlock()
-
 	data := runningTasks.tasks[taskId]
 	if data == nil {
 		return
 	}
 	progressDetail := data.ProgressDetail
 	for p := range progress {
+		runningTasks.mu.Lock()
 		runner.UpdateProgressDetail(db, log, taskId, progressDetail, &p)
+		runningTasks.mu.Unlock()
 	}
 }
 

